boards/sim/sim/sim/scripts/Make.defs

```diff
@@ -93,14 +93,28 @@ NM = $(CROSSDEV)nm
 OBJCOPY = $(CROSSDEV)objcopy
 OBJDUMP = $(CROSSDEV)objdump
 
+# Note: -fshort-wchar for the case where NuttX and the host OS have
+# differnt wchar_t. On Nuttx, it's uint16_t. On macOS, it's 32-bit.
 CFLAGS := $(ARCHWARNINGS) $(ARCHOPTIMIZATION) \
-   $(ARCHCPUFLAGS) $(ARCHINCLUDES) $(ARCHDEFINES) $(EXTRAFLAGS) -pipe
+   $(ARCHCPUFLAGS) $(ARCHINCLUDES) $(ARCHDEFINES) $(EXTRAFLAGS) -pipe \
+   -fshort-wchar
 CXXFLAGS := $(ARCHWARNINGSXX) $(ARCHOPTIMIZATION) \
-   $(ARCHCPUFLAGSXX) $(ARCHXXINCLUDES) $(ARCHDEFINES) $(EXTRAFLAGS) -pipe
-CPPFLAGS := $(ARCHINCLUDES) $(ARCHDEFINES) $(EXTRAFLAGS)
+   $(ARCHCPUFLAGSXX) $(ARCHXXINCLUDES) $(ARCHDEFINES) $(EXTRAFLAGS) -pipe \
+   -fshort-wchar
+CPPFLAGS := $(ARCHINCLUDES) $(ARCHDEFINES) $(EXTRAFLAGS) \
+   -fshort-wchar
 AFLAGS := $(CFLAGS) -D__ASSEMBLY__
 
 ifeq ($(CONFIG_LIBCXX),y)
+  # Why c++14?
+  #  * libcxx seems to require c++11.
+  #  * The compiler defaults varies:
+  #      clang/macOS (from xcode): 199711L
+  #      gcc/ubuntu:               201402L
+  #  * There is a precedent to use c++14.
+  #    (boards/arm/stm32l4/nucleo-l476rg/scripts/Make.defs)
+  CXXFLAGS += -std=c++14 -D_LIBCPP_STD_VER=14
```

Comment:
Don't need _LIBCPP_STD_VER=14, libcxx will detect the right version automatically:
https://github.com/llvm/llvm-project/blob/master/libcxx/include/__config#L45-L57

Reply:
ok